test/integration/targets/git/tasks/archive.yml

```
@@ -74,3 +74,47 @@
   when:
     - "ansible_os_family == 'RedHat'"
     - ansible_distribution_major_version is version('7', '>=')
+
```

We need to add another test to verify the content of the archive. This will catch the current bug that adds a prefix of `None` if `archive` is set but not `archive_prefix`.
```suggestion

- name: ARCHIVE | Inspect archive file
  command:
    cmd: "{{ git_list_commands[item] }} {{ checkout_dir }}/test_role.{{ item }}"
    warn: no
  register: archive_content
  with_items: "{{ git_archive_extensions[ansible_os_family ~ ansible_distribution_major_version | default('default') ] | default(git_archive_extensions.default) }}"

- name: ARCHIVE | Ensure archive content is correct
  assert:
    that:
      - item.stdout_lines | sort | first == 'defaults/'
  with_items: "{{ archive_content.results }}"
```

This is the dict to lookup up the correct list command, which can be added to `vars/main.yml`
```yaml
git_list_commands:
  tar.gz: tar -tf
  tar: tar -tf
  tgz: tar -tf
  zip: unzip -Z1
```